src/core/diags/factory_diags.cpp

```diff
@@ -834,6 +892,45 @@ void Diags::TransmitDone(Error aError)
     return;
 }
 
+bool Diags::AddressMatch(const Mac::Address &aAddressA, const Mac::Address &aAddressB)
+{
+    bool ret = false;
+
+    VerifyOrExit(aAddressA.GetType() == aAddressB.GetType());
+
+    switch (aAddressA.GetType())
+    {
+    case Mac::Address::kTypeNone:
+        ret = true;
+        break;
+
+    case Mac::Address::kTypeShort:
+        VerifyOrExit(aAddressA.GetShort() != aAddressB.GetShort(), ret = true);
+        break;
+
+    case Mac::Address::kTypeExtended:
+        ret = (memcmp(aAddressA.GetExtended().m8, aAddressB.GetExtended().m8, sizeof(Mac::ExtAddress)) == 0);
+        break;
+    }
+
+exit:
+    return ret;
+}
+
+bool Diags::ShouldHandleReceivedFrame(const otRadioFrame &aFrame)
+{
+    bool                ret   = false;
+    const Mac::RxFrame *frame = static_cast<const Mac::RxFrame *>(&aFrame);
```

Comment:
Can consider keeping it a reference?
```suggestion
    const Mac::RxFrame &frame = static_cast<const Mac::RxFrame &>(aFrame);
```